lms/djangoapps/grades/scores.py

```diff
@@ -5,106 +5,242 @@
 
 from openedx.core.lib.cache_utils import memoized
 from xblock.core import XBlock
+from xmodule.block_metadata_utils import display_name_with_default_escaped
+from xmodule.graders import ProblemScore
 from .transformer import GradesTransformer
 
 
 log = getLogger(__name__)
 
 
-@memoized
-def block_types_possibly_scored():
+def possibly_scored(usage_key):
     """
-    Returns the block types that could have a score.
+    Returns whether the given block could impact grading (i.e.
+    has_score or has_children).
+    """
+    return usage_key.block_type in _block_types_possibly_scored()
 
-    Something might be a scored item if it is capable of storing a score
-    (has_score=True). We also have to include anything that can have children,
-    since those children might have scores. We can avoid things like Videos,
-    which have state but cannot ever impact someone's grade.
+
+def get_score(submissions_scores, csm_scores, persisted_block, block):
     """
-    return frozenset(
-        cat for (cat, xblock_class) in XBlock.load_classes() if (
-            getattr(xblock_class, 'has_score', False) or getattr(xblock_class, 'has_children', False)
-        )
+    Returns the score for a problem, as a ProblemScore object.  It is
+    assumed that the provided storages have already been filtered for
+    a single user in question and have user-specific values.
+
+    The score is retrieved from the provided storages in the following
+    order of precedence.  If no value for the block is found in a
+    given storage, the next storage is checked.
+
+    submissions_scores (dict of {unicode(usage_key): (earned, possible)}):
+
+        A python dictionary of serialized UsageKeys to (earned, possible)
+        tuples. These values, retrieved using the Submissions API by the
+        caller (already filtered for the user and course), take precedence
+        above all other score storages.
+
+        When the score is found in this storage, it implies the user's score
+        for the block was persisted via the submissions API. Typically, this API
+        is used by ORA.
+
+        The returned score includes valid values for:
+            w_earned
+            w_possible
+            graded - retrieved from the persisted block, if found, else from
+                the latest block content.
+
+        Note: r_earned and r_possible are not required when submitting scores
+        via the submissions API.  And so those along with weight (since unused)
```

Comment:
Wording nit: `...the submissions API, so those values (along with the unused weight) are invalid and irrelevant.`